pipit/trace.py

```diff
@@ -518,58 +518,118 @@ def comm_by_process(self, output="size"):
             pd.DataFrame: DataFrame containing total communication volume or
             number of messags sent and received by each process.
         """
-        comm_matrix = self.comm_matrix(output=output)
+        comm_matrix = self.comm_mtrix(output=output)
 
         # Get total sent and received for each process
         sent = comm_matrix.sum(axis=1)
         received = comm_matrix.sum(axis=0)
 
         return pd.DataFrame({"Sent": sent, "Received": received}).rename_axis("Process")
 
-    def flat_profile(
-        self, metrics="time.exc", groupby_column="Name", per_process=False
-    ):
+    def flat_profile(self,
+                     metrics: str | list[str] = "time.exc",
+                     groupby_cols: str | list[str] = "Name",
+                     per_process : bool = False,
+                     drop_zeros : bool = False,
+                     ascending : bool = False) -> pd.DataFrame:
         """
+        Generates flat profile of aggregate performance data
```

Comment:
This comment can be more descriptive. We can talk about the aggregations we are doing and how we sort the result.